### PR TITLE
General code maintenance for smart fridge

### LINKED ISSUE #77593: Drying Racks Falsely Inherit Moorings
Reporting client version: 514.1589

## Round ID:
[212360](https://scrubby.melonmesa.com/round/212360)

## Testmerges:
- [Feature: bitrunner, a new supply role](https://www.github.com/tgstation/tgstation/pull/77259)

## Reproduction:
image
because the drier is, I think, a subspecies of the smartfridge, it also gets a descriptor regarding the moorings the fridge gets, even if trying to either "close the maint hatch" or "weld the mooring" does jack all, the only thing that works is the expected wrench for its bolts

### LINKED ISSUE #77805: Invalid list `Find()` call in smartfridge initialization
This line passes the list as a `Start` argument. This doesn't make sense, and BYOND just treats it as 1. This second argument can be removed.

https://github.com/tgstation/tgstation/blob/1e7b41b0f62c662665be7e9dd74746e9b78c6d7c/code/game/objects/items/circuitboards/machines/machine_circuitboards.dm#L546

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
- Fixes #77805
- Fixes #77593

Overall code optimizations for smart fridges. The gist of it includes.
1. Converted UI to typescript
2. Smart fridges won't be accepting items that are either abstract or holograms
3. Autodoc for procs & vars
4. Merged a lot of if statements into 1 ternary operation. Used `is_type_in_list()` in some places to reduce the redundant `istype()` checks
5. Removed all single letter var names
6. Moved all subtype specific vars and functions away from the parent fridge type. If we need a subtype to perform a certain operation the parent type should provide vars & procs to override and we should not be checking for child types inside the parent type. In this case it was overlays for the contents displayed on the fridge & examines

Qol improvements include.
1. Added better examine & screen tips for fridge's
2. Drying rack can now be deconstructed with a crowbar and not simply pried open with it.

## Changelog
:cl:
fix: drying rack now shows correct examines & screen tips.
code: tone of code organization for smart fridges overall. changed ui to typescript.
qol: added more detailed examines & screen tips for smart fridges. drying racks can be dismantled with a crowbar and not simply pried open with it.
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/SmartVend.tsx [hunk 1]
```diff
@@ -19,18 +32,18 @@ export const SmartVend = (props, context) => {
               </Button>
             )
           }>
-          {(data.contents.length === 0 && (
+          {(contents.length === 0 && (
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/SmartVend.tsx:35]:
```suggestion
          {(contents.length === 0 ? (
```

File: tgui/packages/tgui/interfaces/SmartVend.tsx [hunk 2]
```diff
@@ -19,18 +32,18 @@ export const SmartVend = (props, context) => {
...
             <NoticeBox>Unfortunately, this {data.name} is empty.</NoticeBox>
           )) || (
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/SmartVend.tsx:37]:
```
)) : (
```

**Comment by SyncIt21** [comment]
I don't know what about the map file changes. Those are not mine and i don't know how to get rid of them.

**Comment by SyncIt21** [comment]
@jlsnow301 any more reviews?

**Pull request merged**